src/test/java/de/tum/in/www1/artemis/ModelingExerciseIntegrationTest.java

```diff
@@ -539,6 +539,34 @@ void testInstructorGetsOnlyResultsFromOwningCourses() throws Exception {
         assertThat(result.getResultsOnPage()).isNullOrEmpty();
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    void testInstructorSearchTermMatchesId() throws Exception {
+        database.resetDatabase();
+        database.addUsers(1, 1, 0, 1);
+        testSearchTermMatchesId();
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void testAdminSearchTermMatchesId() throws Exception {
+        database.resetDatabase();
+        database.addUsers(1, 1, 0, 1);
+        testSearchTermMatchesId();
+    }
+
+    private void testSearchTermMatchesId() throws Exception {
+        final Course course = database.addEmptyCourse();
+        final var now = ZonedDateTime.now();
+        ModelingExercise exercise = ModelFactory.generateModelingExercise(now.minusDays(1), now.minusHours(2), now.minusHours(1), DiagramType.ClassDiagram, course);
+        exercise.setTitle("LoremIpsum");
+        exercise = modelingExerciseRepository.save(exercise);
+
+        final var searchTerm = database.configureSearch("" + exercise.getId());
```

Comment:
```suggestion
        final var searchTerm = database.configureSearch(exercise.getId().toString());
```
Codacy complains about the Stringconcatenation

Reply:
Done in d68b55f50edde75142d8cb362c9a6401a2c51ba8